### PR TITLE
ENH: add new wheel builds using Accelerate on macOS >=14

### PR DESCRIPTION
**Pull request by andyfaff**
@rgommers alternative to #25012.

EDIT: the new Accelerate-based wheels weigh in at about 5.0 MB, which is impressively small compared to the OpenBLAS arm64 wheels at ~14 MB and the OpenBLAS x86-64 wheels at ~21 MB.

### CODE REVIEW

**Comment by andyfaff** [comment]
This should work now. There are two cpN wheels produced for macos_arm64. Those suitable for macOS < 14 and those for macOS >= 14.0.
The linux_aarch64 is currently disabled, and should be renabled before merge.

**Comment by rgommers** [comment]
Thanks @andyfaff! This looks pretty good to me, however it's not a complete replacement for gh-25012 and lost a few things from that PR. How about I resubmit the commits that were good to go from that PR separately, we merge those, and then we rebase this PR on top? That will then be a slightly smaller diff here with only the essential changes for Accelerate wheels.

**Comment by rgommers** [comment]
This still looks good. I went through the Python 3.12 log in detail just now, and everything looks as expected. Capturing some details for future reference:

```
cibuildwheel version 2.16.2

Project version: 2.0.0.dev0+git20231126.c6bcaaa
C compiler for the host machine: arm64-apple-darwin20.0.0-clang (clang 16.0.6 "clang version 16.0.6")
C linker for the host machine: arm64-apple-darwin20.0.0-clang ld64 609
C++ compiler for the host machine: arm64-apple-darwin20.0.0-clang++ (clang 16.0.6 "clang version 16.0.6")
C++ linker for the host machine: arm64-apple-darwin20.0.0-clang++ ld64 609
Cython compiler for the host machine: cython (cython 3.0.7)
Host machine cpu family: aarch64
Host machine cpu: aarch64
Program python found: YES (/private/var/folders/c2/b2kmn__12wq57m82ltz0hl1w0000gn/T/build-env-sqihr22c/bin/python)
Found pkg-config: /opt/homebrew/bin/pkg-config (0.29.2)
Run-time dependency python found: YES 3.12
Has header "Python.h" with dependency python-3.12: YES 
Compiler for C supports arguments -fno-strict-aliasing: YES 
Compiler for C supports arguments -ftrapping-math: YES 
Compiler for C supports link arguments -Wl,-ld_classic: NO 
Message: During parsing cpu-dispatch: The following CPU features were ignored due to platform incompatibility or lack of support:
"XOP FMA4"
Test features "NEON NEON_FP16 NEON_VFPV4 ASIMD" : Supported 
Test features "ASIMDHP" : Supported 
Test features "ASIMDFHM" : Supported 
Test features "SVE" : Unsupported due to Compiler fails against the test code of "SVE"

Run-time dependency scipy-openblas found: NO (tried pkgconfig)
Run-time dependency mkl found: NO (tried pkgconfig and system)
Run-time dependency accelerate found: YES
../numpy/meson.build:111: WARNING: Project targets '>=1.2.99' but uses feature introduced in '1.3.0': dep 'accelerate' custom lookup.
Message: BLAS symbol suffix: $NEWLAPACK$ILP64

1 wheels produced in 4 minutes:
  numpy-2.0.0.dev0-cp312-cp312-macosx_14_0_arm64.whl   4,725 kB
```

**Comment by rgommers** [comment]
Compared to the `11_0` wheel with OpenBLAS:
```
1 wheels produced in 4 minutes:
  numpy-2.0.0.dev0-cp312-cp312-macosx_11_0_arm64.whl   13,398 kB
```

That's an almost 3x size reduction.

**Review by rgommers — APPROVED** [review]
This LGTM, and everything seems happy. The rerun of the last wheel build didn't crash, so almost certainly related to `sort` and gh-25445. I'll aim to merge this once the `aarch64` wheel builds are fixed (there won't be uploads from the new jobs before then).

**Comment by rgommers** [comment]
Actually, the `memory: 4G` fix in here is needed for the aarch64 builds, so let me merge it now. Thanks again @andyfaff!

**Pull request merged**
